### ISSUE TITLE
About reproducing results on SemanticKITTI

### ISSUE THREAD

**Issue by lwwdww** [USTCLH/OccMamba#11]
Thanks for your great work! I have followed the dataset config for SemanticKITTI as mentioned in https://github.com/USTCLH/OccMamba/issues/3, but the reproduced results as follows are severely lower than the reported results.

```
|     class     |  IoU  |
+---------------+-------+
|      free     | 0.926 |
|      car      | 0.216 |
|    bicycle    | 0.003 |
|   motorcycle  | 0.003 |
|     truck     | 0.023 |
| other-vehicle | 0.014 |
|     person    | 0.011 |
|   bicyclist   | 0.015 |
|  motorcyclist |  0.0  |
|      road     | 0.549 |
|    parking    | 0.152 |
|    sidewalk   | 0.249 |
|  other-ground | 0.002 |
|    building   | 0.159 |
|     fence     | 0.039 |
|   vegetation  | 0.196 |
|     trunk     | 0.025 |
|    terrain    | 0.304 |
|      pole     | 0.039 |
|  traffic-sign | 0.022 |
|      mean     | 0.106 |
+---------------+-------+
```

Could you please provide the training and evaluation pipelines for SemanticKITTI dataset?

**Comment by USTCLH** [comment]
Apologies for the delay. Since this project was completed some time ago and I experienced a hard drive wipe, I have to rely on my memory and some remaining notes to provide a solution. The code in this repository is primarily tuned for the OpenOccupancy dataset. Indeed, the configurations for SemanticKITTI are quite different. If you apply the following settings, you should see a significant improvement in performance.

```
pts_voxel_layer=dict(
        num_point_features=4,
        max_num_points=10, 
        point_cloud_range=point_cloud_range,
        voxel_size=[0.1, 0.1, 0.1],
        max_voxels=(90000, 120000)),
    pts_voxel_encoder=dict(
        type='cylinder_fea', 
        num_features=4,
        out_pt_fea_dim=256,
        fea_compre=32,
    ),
    pts_middle_encoder=dict(
        type='Asymm_3d_spconv',
        in_channels=32,
        out_channels=numC_Trans,
        sparse_shape=[64, 512, 512],
        downsample_layers=2,
        init_size=32,
    ),
```
SemanticKITTI 64-beam LiDAR is significantly denser than the 32-beam Lidar used in OpenOccupancy. Using cylindrical coordinates seems better captures the native geometry in SemanticKITTI.

```
bda_aug_conf = dict(
    rot_lim=(0, 0),
    scale_lim=(1.0, 1.0),
    flip_dx_ratio=0.5,
    flip_dy_ratio=0.5,
    flip_dz_ratio=0,)
```
Empirical results suggest that heavy geometric augmentation (like dz flip) in the BEV space can sometimes hinder convergence or introduce noise, especially when the voxel resolution is high and using Mamba. Simplifying the augmentation pipeline stabilizes the training.

Additionally, I recommend lowering the learning rate (e.g., try `3e-4`).

**Comment by lwwdww** [comment]
Thanks very much! The reproduced result on SemanticKITTI achieves 20 mIoU now, but still has a gap between the reperted result. I notice that the paper mentioned "Test Time Augmentation" for SemanticKITTI dataset, is this the reason that the reproduced results are lower the the reported results? Could you provided the implementation of the "Test Time Augmentation"?

**Comment by USTCLH** [comment]
The "Test Time Augmentation" (TTA) implementation is as follows: Since we applied flips along the x and y axes during training, we do something similar during inference. For each input, we create 4 variations corresponding to all combinations of x and y flips. Then, across these 4 outputs, we average the probabilities of each voxel (from the layer before the final prediction) to obtain the final output. However, please note that applying this TTA trick did not bring a significant improvement in mIoU.

Additionally, when I was trying to improve the mIoU on the SemanticKITTI dataset back then, I actually spent a long time tuning various hyperparameters. If you are struggling to get the desired results in a short time, I wouldn't recommend spend much more time into tuning it.

The core design philosophy of OccMamba is geared towards handling larger scenes and higher-resolution Occupancy Prediction tasks (such as the OpenOccupancy benchmark). In relatively smaller scenes and lower voxels like SemanticKITTI, its advantages are quite marginal compared to other methods that rely on dense processing. This is also reflected in our paper, as you can see by comparing the mIoU gaps between our method and the second-place methods across the two datasets.
